Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -8,37 +11,60 @@ import { ArtifactItemStore } from '../Store/artifactItemStore';
 import { ArtifactEngineOptions } from "./artifactEngineOptions"
 import { Logger } from './logger';
 import { Worker } from './worker';
+import { TicketState } from '../Models/ticketState';
+import { CacheProvider } from '../Providers/cacheProvider';
 
 export class ArtifactEngine {
     processItems(sourceProvider: models.IArtifactProvider, destProvider: models.IArtifactProvider, artifactEngineOptions?: ArtifactEngineOptions): Promise<models.ArtifactDownloadTicket[]> {
         var artifactDownloadTicketsPromise = new Promise<models.ArtifactDownloadTicket[]>((resolve, reject) => {
             const workers: Promise<void>[] = [];
             artifactEngineOptions = artifactEngineOptions || new ArtifactEngineOptions();
+            artifactEngineOptions.artifactCacheDirectory = artifactEngineOptions.artifactCacheDirectory ? artifactEngineOptions.artifactCacheDirectory : tl.getVariable("AGENT_WORKFOLDER");
             this.createPatternList(artifactEngineOptions);
+            var artifactName = sourceProvider.getRootItemPath();
+            this.artifactItemStore = new ArtifactItemStore(artifactName);
             this.artifactItemStore.flush();
             Logger.verbose = artifactEngineOptions.verbose;
             this.logger = new Logger(this.artifactItemStore);
             this.logger.logProgress();
             sourceProvider.artifactItemStore = this.artifactItemStore;
             destProvider.artifactItemStore = this.artifactItemStore;
+            if (!artifactEngineOptions.enableIncrementalDownload) {
+                artifactEngineOptions.artifactCacheDirectory = '';
+                artifactEngineOptions.artifactCacheKey = '';
+            }
+            this.cacheProvider = new CacheProvider(artifactEngineOptions.artifactCacheDirectory, artifactEngineOptions.artifactCacheKey, artifactName);
```

Comment:
looks like we are creating cacheProvider even if artifactEngineOptions.enableIncrementalDownload is false. correct? 

is the intent to avoid checks for artifactEngineOptions.enableIncrementalDownload in multiple places?

are we going to copy over files downloaded from server back to cache at the end if artifactEngineOptions.enableIncrementalDownload is false?

Reply:
yes! cacheProvider is created but both the hashMaps are returned as empty so normal download process that was implemented before runs without making or looking at the cache.

No files are not copied if artifactEngineOptions.enableIncrementalDownload is false.